utilities/titandb/titan_db_test.cc

```diff
@@ -237,19 +298,56 @@ TEST_F(TitanDBTest, DbIter) {
 }
 
 TEST_F(TitanDBTest, Snapshot) {
-    Open();
-    std::map<std::string, std::string> data;
-    Put(1, &data);
-    ASSERT_EQ(1, data.size());
+  Open();
+  std::map<std::string, std::string> data;
+  Put(1, &data);
+  ASSERT_EQ(1, data.size());
 
-    const Snapshot* snapshot(db_->GetSnapshot());
-    ReadOptions ropts;
-    ropts.snapshot = snapshot;
+  const Snapshot* snapshot(db_->GetSnapshot());
+  ReadOptions ropts;
+  ropts.snapshot = snapshot;
 
-    VerifyDB(data, ropts);
-    Flush();
-    VerifyDB(data, ropts);
-    db_->ReleaseSnapshot(snapshot);
+  VerifyDB(data, ropts);
+  Flush();
+  VerifyDB(data, ropts);
+  db_->ReleaseSnapshot(snapshot);
+}
+
+TEST_F(TitanDBTest, IngestExternalFiles) {
+  Open();
+  SstFileWriter sst_file_writer(EnvOptions(), options_);
+  ASSERT_EQ(sst_file_writer.FileSize(), 0);
+
+  const uint64_t kNumEntries = 100;
+  std::map<std::string, std::string> data;
+  for (uint64_t i = 1; i <= kNumEntries; i++) {
+    Put(i, &data);
+  }
+  ASSERT_EQ(kNumEntries, data.size());
+  Flush();
+  VerifyDB(data);
+
+  const uint64_t kNumIngestedEntries = 100;
+  std::string sst_file = options_.dirname + "/for_ingest.sst";
+  ASSERT_OK(sst_file_writer.Open(sst_file));
+  for (uint64_t i = 1; i <= kNumIngestedEntries; i++) {
+    std::string key = GenKey(kNumEntries + i);
```

Comment:
you may want to have the keys in the sst overlap with the keys inserted before. otherwise rocksdb may just do "trivial move" and refuse to run actual compaction and break our test. For example, the following can happen:
1. we insert keys [a-b]
2. we ingest file with keys [c-d]
2.1. rocksdb flush the existing keys into one file with [a-b] before ingestion
2.2. rocksdb ingest the file [c-d]
3. we issue compact range. since [a-b] and [c-d] don't overlap, rocksdb may just trivially move the two files to bottom level in the LSM tree. since compaction is not actually running over the ingested file, large value in the ingested file is not written to blob file.